### PR TITLE
Sdev codesplitting

### PR DESCRIPTION
**Pull request by TheAlaskanPhysicist**
Changed the structure of procedurally generated blocks into separate functions to prepare for removal of implicit function calls like GTItems does with its procedurally generated blocks.

### CODE REVIEW

File: common/src/main/java/com/gregtechceu/gtceu/common/data/GTBlocks.java [hunk 1]
```diff
@@ -98,267 +98,360 @@
 public class GTBlocks {
 
     //////////////////////////////////////
-    //*****     Material Blocks    *****//
+    //*****     Tables Builders    *****//
     //////////////////////////////////////
+    // TODO: Remove builders after initialization
```
Comment (mikerooni) [inline common/src/main/java/com/gregtechceu/gtceu/common/data/GTBlocks.java]: Not 100% sure whether they actually need to be fields - might be able to keep them local in the methods

File: common/src/main/java/com/gregtechceu/gtceu/common/data/GTBlocks.java [hunk 2]
```diff
@@ -98,267 +98,360 @@
...
-                        .build()
-                        .register();
-                builder.put(TagPrefix.block, material, entry);
+            if (allowMaterialBlock(material)) {
+                registerMaterialBlock(material);
             }
+        }
+        GTCEu.LOGGER.info("Generating Compressed Blocks... Complete!");
+    }
+    public static boolean allowMaterialBlock(Material material) {
```
Comment (mikerooni) [inline common/src/main/java/com/gregtechceu/gtceu/common/data/GTBlocks.java]: The `allow...()` and `register...()` methods should be private

File: common/src/main/java/com/gregtechceu/gtceu/common/data/GTBlocks.java [hunk 3]
```diff
@@ -98,267 +98,360 @@
...
+                .defaultLoot()
+                .tag(GTToolType.WIRE_CUTTER.harvestTag)
+                .addLayer(() -> RenderType::cutoutMipped)
+                .color(() -> LaserPipeBlock::tintedColor)
+                .item(LaserPipeBlockItem::new)
+                .model(NonNullBiConsumer.noop())
+                .color(() -> LaserPipeBlockItem::tintColor)
+                .build()
+                .register();
+        LASER_PIPES[slot] = entry;
```
Comment (mikerooni) [inline common/src/main/java/com/gregtechceu/gtceu/common/data/GTBlocks.java:449]: Might be better to return the entry here and assign it in the calling function, could possibly even use something like this if you change the param type to `LaserPipeType`:

```java
LASER_PIPES = Arrays.stream(LaserPipeType.values())
    .map(GTBlocks::registerLaserPipeBlock)
    .toArray(BlockEntry[]::new)
```

**Review by mikerooni — COMMENTED** [review]
Looks great so far! Got a few more suggestions ontop of it as well ^^

File: common/src/main/java/com/gregtechceu/gtceu/common/data/GTBlocks.java (hunk 2 above)
Comment (TheAlaskanPhysicist) [reply common/src/main/java/com/gregtechceu/gtceu/common/data/GTBlocks.java]: Resolved in next push

File: common/src/main/java/com/gregtechceu/gtceu/common/data/GTBlocks.java (hunk 1 above)
Comment (TheAlaskanPhysicist) [reply common/src/main/java/com/gregtechceu/gtceu/common/data/GTBlocks.java]: These will be depreciated in effect when the initialization process is abstracted like GTItems

**Comment by TheAlaskanPhysicist** [comment]
I believe this section to be done. Lmk if anyone has other suggestions.

**Review by mikerooni — APPROVED** [review]
Looks good ^^

**Comment by mikerooni** [comment]
Only suggestion I'd have is maybe replacing the table builders with streams if possible, but that's kinda optional tbh

**Comment by TheAlaskanPhysicist** [comment]
> Only suggestion I'd have is maybe replacing the table builders with streams if possible, but that's kinda optional tbh

I'll open another branch when I can get it successfully. It's a little jank but maybe I'll have it done by new years.

**Pull request merged**